build_tools/scripts/update_submodules.sh

```diff
@@ -37,8 +37,29 @@ if [ -n "$git_status" ]; then
   exit 1
 fi
 
-echo "Pulling latest from remote"
-git pull
+master_url="$(git remote get-url master)"
+upstream_url="https://github.com/google/iree.git"
+if [ ${master_url} == ${upstream_url} ]
+then
+  echo "Pulling latest from remote"
+  git pull
+else
+  for remote in `git remote -v | awk -F\  '{print $1}' | sort | uniq`
+  do
+    remote_url="$(git remote get-url ${remote})"
+    if [ ${remote_url} == ${upstream_url} ]
+    then
+      upstream=${remote}
+      break
+    fi
+  done
+
+  echo "Fetching" ${upstream}
+  git fetch ${upstream}
+
+  echo "Merging fork master with " ${upstream}
+  git merge ${upstream}/master
```

Comment:
Might rebasing be the preferred way to go?

Reply:
Yeah, probably rebase is better. Ill update and try it.

Reply:
I think this whole thing could be

`git pull --rebase https://github.com/google/iree.git`

Reply:
This is done now. THanks for the tip!